src/values/ObjectValue.js

```diff
@@ -404,16 +408,70 @@ export default class ObjectValue extends ConcreteValue {
     return this._isFinal.mightNotBeTrue();
   }
 
-  havoc(): void {
-    this._isHavoced = this.$Realm.intrinsics.true;
+  leak(): void {
+    this._isLeaked = this.$Realm.intrinsics.true;
   }
 
-  mightBeHavocedObject(): boolean {
-    return this._isHavoced.mightBeTrue();
+  /* Emits generator entries to set the values of properties in an object */
+  materializeInGenerator(generator: Generator, getHeapInspector?: void => HeapInspector, emitDeletes?: boolean): void {
```

Comment:
This seems to do the same kind of thing as ObjectValue.getSnapSnot. That seems like a great pity to me.

Reply:
The great pity to me is that ObjectValue.getSnapshot is not documented, as is not the model of which it is a part. Nonetheless, I will see if I can somehow decipher its behavior and reconcile it with my changes.

Reply:
@sb98052 I can fill you in on how getSnapshot works if that helps? Feel free to ping me and we can setup a VC or chat over workplace. Otherwise, I'll be sure to put up a PR tomorrow adding some comments around it if that helps.

Reply:
Thanks. Comments would certainly be helpful. Right now, the `materializeInGenerator` function factorizes out code that was already in the (old-terminology-) havocing implementation. As such, if there is any redundancy between these two things, then it is pre-existing, and out of scope for this pull request. I would be happy to take on the task of reconciling the two things in a separate task.